donation-tracker/static/api/index.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * Class to define timestamp notes
+ */
+class Notes {
+  public $lastRead = '';
+  public $lastDataUpdate = '';
+}
+
+/**
+ * Class with the main buffer script
+ */
+class DataBuffer {
+  private $remoteCsvDataSource;
+  private $localNotesFile;
+  private $localBufferFile;
+  private $remoteCheckInterval;
+
+  /**
+   * Create a new instance of the DataBuffer
+   */
+  function __construct(
+      $remoteCsvDataSource,
+      $localNotesFile,
+      $localBufferFile,
+      $remoteCheckInterval
+    ) {
+    $this->remoteCsvDataSource = $remoteCsvDataSource;
+    $this->localNotesFile = $localNotesFile;
+    $this->localBufferFile = $localBufferFile;
+    $this->remoteCheckInterval = $remoteCheckInterval;
+  }
+
+  /**
+   * Function to get the latest CSV data from local buffer
+   */
+  public function getCsvData() {
+    // this is a small wrapper that runs server-side to reduce the Google API calls
+    $notes = new Notes();
+    $dataToServe = '';
+    try {
+      if (file_exists($this->localNotesFile) && file_exists($this->localBufferFile)) {
+        // we have a local buffer on the server side
+        $notes = json_decode(file_get_contents($this->localNotesFile));
+        if (($notes->lastRead + $this->remoteCheckInterval) < time()) {
+          // wait time is over - we need to check remote
+          $tmpLocalData = file_get_contents($this->localBufferFile);
+          $tempNewData = file_get_contents($this->remoteCsvDataSource);
+          if ($tempNewData === null) {
+            // in case we do not get anything from remote server: use local data
+            $dataToServe = $tmpLocalData;
+            // remember check and persist the timestamp
+            $this->updateLastRemoteCheckTimestamp($notes);
+          } else {
+            $lastDataUpdate = $this->extractDataUpdateTime($tempNewData);
+            if ($lastDataUpdate > $notes->lastDataUpdate) {
+              // use newer data and buffer it
+              $dataToServe = $tmpLocalData;
```

Comment:
I guess this should use and store the new data instead:
```suggestion
              $dataToServe = $tempNewData;
```